src/sentry/users/api/endpoints/user_details.py

```diff
@@ -230,15 +249,33 @@ def put(self, request: Request, user: User) -> Response:
             "defaultIssueEvent": "default_issue_event",
             "clock24Hours": "clock_24_hours",
             "prefersIssueDetailsStreamlinedUI": "prefers_issue_details_streamlined_ui",
+            "quickStartDisplay": "quick_start_display",
         }
 
         options_result = serializer_options.validated_data
 
         for key in key_map:
             if key in options_result:
-                UserOption.objects.set_value(
-                    user=user, key=key_map.get(key, key), value=options_result.get(key)
-                )
+                if key == "quickStartDisplay":
+                    current_value = UserOption.objects.get_value(
+                        user=user, key=key_map.get(key, key)
+                    )
+
+                    if current_value is None:
+                        current_value = {}
+
+                    new_value = options_result.get(key)
+
+                    for org_id, display_value in new_value.items():
+                        current_value[org_id] = display_value
```

Comment:
If you have two dicts, you can do this, it will set all keys present in `new_value`, and leave the other keys & values present only in `current_value` intact: 
```suggestion
                    current_value.update(new_value)
```